### PR TITLE
mach: fix test-speedometer

### PR DESCRIPTION
**Pull request by sagudev**
Split from https://github.com/servo/servo/pull/33261

---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes do not require tests because they are for tests

### CODE REVIEW

File: python/servo/testing_commands.py [hunk 1]
```diff
@@ -414,6 +414,22 @@ def test_dromaeo(self, tests, servo_binary: str, bmf_output: str | None = None):
...
+        if bmf_output:
+            with open(bmf_output, 'w', encoding='utf-8') as f:
+                json.dump({
+                    'servo': {
+                        'file-size': {
+                            'value': float(size),
+                        }
+                    }
+                }, f, indent=4)
+
```
Comment (mrobinson) [inline python/servo/testing_commands.py:432]: Is the BMF file format defined somewhere?

File: python/servo/devenv_commands.py [hunk 2]
```diff
@@ -270,3 +271,18 @@ def ndk_gdb(self, release, target):
...
+        output: dict[str, object] = dict()
+        for input_file in inputs:
+            with open(input_file, 'r', encoding='utf-8') as f:
+                output = json.load(f) | output
+
+        if bmf_output:
+            with open(bmf_output, 'w', encoding='utf-8') as f:
+                json.dump(output, f, indent=4)
+        else:
+            print(json.dumps(output, indent=4))
```
Comment (mrobinson) [inline python/servo/devenv_commands.py:288]: Where is this merge command used? It feels almost a bit too specific for a particular task and also this is a few lines of Python. It's hard to know if this is appropriate for `mach` without having the full context.

**Review by mrobinson — CHANGES_REQUESTED** [review]
Can you remove the two new commands and just fix `speedometer_runner` in this PR?

File: python/servo/testing_commands.py (hunk 1 above)
Comment (sagudev) [reply python/servo/testing_commands.py:432]: https://bencher.dev/docs/reference/bencher-metric-format/

File: python/servo/devenv_commands.py (hunk 2 above)
Comment (sagudev) [reply python/servo/devenv_commands.py:288]: TBH I was also thinking the same thing, maybe this and test-size should rather go to etc/ci a part of https://github.com/servo/servo/pull/33261

Comment (mrobinson) [reply python/servo/devenv_commands.py:288]: Yeah, that seems reasonable! That way the scripts will be available if / when we stop using `mach`.

**Review by mrobinson — APPROVED** [review]

**Pull request merged**